lib/analyse.ml

```diff
@@ -163,44 +161,51 @@ module Analysis = struct
     | Error (`Msg msg) -> Fmt.error_msg "Error from solver: %s" msg
 
   let of_dir ~solver ~job ~platforms ~opam_repository_commit dir =
-    let solve = solve ~opam_repository_commit ~job ~solver in
-    let ty = type_of_dir dir in
-    let cmd = "", [| "find"; "."; "-maxdepth"; "3"; "-name"; "*.opam" |] in
-    Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>!= fun output ->
     let opam_files =
-      String.split_on_char '\n' output
-      |> List.sort String.compare
-      |> List.filter_map (function
-          | "" -> None
-          | path ->
-            let path =
-              if Astring.String.is_prefix ~affix:"./" path then
-                Astring.String.with_range ~first:2 path
-              else path
-            in
-            let consider_opam_file path =
-              match Fpath.v path |> Fpath.segs with
-              | [_file] -> true
-              | segs when List.exists is_test_dir segs ->
-                Current.Job.log job "Ignoring test directory %S" path;
-                false
-              | _ -> true
-            in
-            let full_path = Filename.concat (Fpath.to_string dir) path in
-            if is_empty_file full_path then (
-              Current.Job.log job "WARNING: ignoring empty opam file %S" path;
-              None
-            ) else if consider_opam_file path then
-              Some path
-            else None
-        )
+      let opam_files full_path =
+        let segs = match Fpath.segs full_path with "./" :: tl -> tl | l -> l in
+        let path = Fpath.(to_string (Option.get (rem_prefix dir full_path))) in
```

Comment:
Same question as before about the Option.get -- you seem to later on have an error messsage when Fpath.rem_prefix is None, but here it would just leak an error. Can you perhaps move `path` down to where it's used in the function and have the same Error return?

(the use of `dir`, `path`, `full_path` are all a little non-descriptive to me -- perhaps a comment noting that the input of the function is intended to be a relative path would be helpful)

Reply:
I've fixed and simplified the code, I think.
Here the `Option.get` is valid because we start the fold from `dir` and from what I can see it's always the prefix of visited paths.
I've changed the variables: `root` is always the root from where the search is done, `path` the current path, and `test_path` the current test, so that we have `root / test_path = path`. I think it's clearer.

I may need to test this patch further, so I'm leaving it as draft for now.